app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -302,13 +291,34 @@ public function createOrder($reference, $session_quote_id = null)
 
         $this->validateSubmittedOrder($order, $quote);
 
-        // deactivate quote
-        $quote->setIsActive(false);
-        $quote->save();
-
         Mage::getModel('boltpay/payment')->handleOrderUpdate($order);
 
-        Mage::dispatchEvent('bolt_boltpay_save_order_after', array('order'=>$order, 'quote'=>$quote));
+        Mage::dispatchEvent('bolt_boltpay_save_order_after', array('order'=>$order, 'quote'=>$quote, 'transaction' => $transaction));
+
+        if ($sessionQuoteId) {
+            $checkout_session = Mage::getSingleton('checkout/session');
+
+            $checkout_session
+                ->clearHelperData();
+
+            $checkout_session
+                ->setLastQuoteId($quote->getId())
+                ->setLastSuccessQuoteId($quote->getId());
+
+            // add order information to the session
+            $checkout_session->setLastOrderId($order->getId())
+                ->setRedirectUrl('')
+                ->setLastRealOrderId($order->getIncrementId());
+        }
+
+        // Close out session by deleting the parent quote and deactivating the immutable quote so that it can no
+        // longer be used.
+        /* @var Mage_Sales_Model_Quote $parentQuote */
+        $parentQuote = Mage::getModel('sales/quote')->loadByIdWithoutStore($quote->getParentQuoteId());
+        $parentQuote->delete();
+        $quote->setIsActive(false)
+            ->setBoltOrderToken(null)
```

Comment:
why we need to nullify orderToken? Is it useful for debugging if we keep it as is?

Reply:
I was hoping to use the order token as a nonce.  However, it appears that the order token returned by calling the Bolt Create Order API is different than the one returned by calling the Bolt Fetch API.  If this is true, it will be removed in entirety.